pkgs/development/compilers/rust/rustc.nix

```diff
@@ -37,6 +37,9 @@ stdenv.mkDerivation {
   # The build will fail at the very end on AArch64 without this.
   dontUpdateAutotoolsGnuConfigScripts = if stdenv.isAarch64 then true else null;
 
+  # Running the default `strip -S` command corrupts the .rlib files.
+  dontStrip = if stdenv.isDarwin then true else null;
```

Comment:
How about `stripDebugList = stdenv.lib.optional stdenv.isDarwin "bin";` instead of disabling it entirely.

Reply:
It's probably still desirable to strip the regular libraries in `lib`, as well as `bin`. We could copy the logic from strip.sh and modify it to skip `.rlib` files on all platforms:
```nix
{
stripDebugList = [ "bin" ];
preInstall = ''
  find lib -type f ! -name '*.rlib' -print0 | xargs -0 ${xargsFlags:--r} $stripCmd $commonStripFlags ${stripDebugFlags:--S} 2>/dev/null || true
'';
}
```
Or even modify strip.sh to have an extra filter argument?

Happy to go with any recommendation here. Just so long as `.rlib` files are excluded.

Reply:
Yeah, there's currently no nice way to exclude things. Adding something like `stripExcludes` to the stdenv seems like the nicest solution.

Reply:
The change you suggested (restricting to only "bin" on Darwin) seems like the simplest way to fix this problem. Should we merge that to fix rust now, and then later see about adding exclusion to stdenv?